### ISSUE TITLE
Regular expression not terminating on '$'

### ISSUE THREAD

**Issue by johnramsden** [fail2ban/fail2ban#2745]
### Environment:

- Fail2Ban version (including any possible distribution suffixes): `Fail2Ban v0.10.2`
- OS, including release name/version: Ubuntu 18.04
- [x] Fail2Ban installed via OS/distribution mechanisms
- [x] You have not applied any additional foreign patches to the codebase
- [x] Some customizations were done to the configuration (provide details below is so)

### The issue:

What if my actions does not appear to be terminating on `$` and I am getting errors implying many lines of log are attempting to be matched against.

#### Steps to reproduce

Run with systemd haproxy configuration.

#### Expected behavior

Regular expression stops matching on newline.

#### Observed behavior

The log shows many lines attempting to be matched against.

### Configuration, dump and another helpful excerpts

```
['add', 'haproxy-badbots-1440', 'systemd']                                                                                          
['set', 'haproxy-badbots-1440', 'addfailregex', '^\\s*\\w+\\s+haproxy\\[\\d+\\]: <ADDR>.*(?:Atomic_Email_Hunter/4\\.0|atSpider/1\\.0
|autoemailspider|bwh3_user_agent|China Local Browse 2\\.6|ContactBot/0\\.2|ContentSmartz|DataCha0s/2\\.0|DBrowse 1\\.4b|DBrowse 1\\.
4d|Demo Bot DOT 16b|Demo Bot Z 16b|DSurf15a 01|DSurf15a 71|DSurf15a 81|DSurf15a VA|EBrowse 1\\.4b|Educate Search VxB|EmailSiphon|Ema
ilSpider|EmailWolf 1\\.00|ESurf15a 15|ExtractorPro|Franklin Locator 1\\.8|FSurf15a 01|Full Web Bot 0416B|Full Web Bot 0516B|Full Web
 Bot 2816B|Guestbook Auto Submitter|Industry Program 1\\.0\\.x|ISC Systems iRc Search 2\\.1|IUPUI Research Bot v 1\\.9a|LARBIN-EXPER
IMENTAL \\(efp@gmx\\.net\\)|LetsCrawl\\.com/1\\.0 +http\\://letscrawl\\.com/|Lincoln State Web Browser|LMQueueBot/0\\.2|LWP\\:\\:Sim
ple/5\\.803|Mac Finder 1\\.0\\.xx|MFC Foundation Class Library 4\\.0|Microsoft URL Control - 6\\.00\\.8xxx|Missauga Locate 1\\.0\\.0
|Missigua Locator 1\\.9|Missouri College Browse|Mizzu Labs 2\\.2|Mo College 1\\.9|MVAClient|Mozilla/2\\.0 \\(compatible; NEWT Active
X; Win32\\)|Mozilla/3\\.0 \\(compatible; Indy Library\\)|Mozilla/3\\.0 \\(compatible; scan4mail \\(advanced version\\) http\\://www\
\.peterspages\\.net/?scan4mail\\)|Mozilla/4\\.0 \\(compatible; Advanced Email Extractor v2\\.xx\\)|Mozilla/4\\.0 \\(compatible; Iple
xx Spider/1\\.0 http\\://www\\.iplexx\\.at\\)|Mozilla/4\\.0 \\(compatible; MSIE 5\\.0; Windows NT; DigExt; DTS Agent|Mozilla/4\\.0 e
fp@gmx\\.net|Mozilla/5\\.0 \\(Version\\: xxxx Type\\:xx\\)|NameOfAgent \\(CMS Spider\\)|NASA Search 1\\.0|Nsauditor/1\\.x|PBrowse 1\
\.4b|PEval 1\\.4b|Poirot|Port Huron Labs|Production Bot 0116B|Production Bot 2016B|Production Bot DOT 3016B|Program Shareware 1\\.0\
\.2|PSurf15a 11|PSurf15a 51|PSurf15a VA|psycheclone|RSurf15a 41|RSurf15a 51|RSurf15a 81|searchbot admin@google\\.com|ShablastBot 1\\
.0|snap\\.com beta crawler v0|Snapbot/1\\.0|Snapbot/1\\.0 \\(Snap Shots&#44; +http\\://www\\.snap\\.com\\)|sogou develop spider|Sogou Orion spider/3\\.0\\(+http\\://www\\.sogou\\.com/docs/help/webmasters\\.htm#07\\)|sogou spider|Sogou web spider/3\\.0\\(+http\\://www\\.sogou\\.com/docs/help/webmasters\\.htm#07\\)|sohu agent|SSurf15a 11 |TSurf15a 11|Under the Rainbow 2\\.2|User-Agent\\: Mozilla/4\\.0 \\(compatible; MSIE 6\\.0; Windows NT 5\\.1\\)|VadixBot|WebVulnCrawl\\.unknown/1\\.0 libwww-perl/5\\.803|Wells Search II|WEP
Search 00|Seekport Crawler|EmailCollector|WebEMailExtrac|TrackBack/1\\.02|sogou music spider|SEMrushBot|BLEXBot|Python/3.8 aiohttp/3.6.2).*$']
['set', 'haproxy-badbots-1440', 'addjournalmatch', '_SYSTEMD_UNIT=haproxy.service', '+', '_COMM=haproxy']
['set', 'haproxy-badbots-1440', 'logencoding', 'auto']
['set', 'haproxy-badbots-1440', 'maxretry', 6]
['set', 'haproxy-badbots-1440', 'findtime', '7h']
['set', 'haproxy-badbots-1440', 'bantime', '24h']
['set', 'haproxy-badbots-1440', 'usedns', 'warn']
['set', 'haproxy-badbots-1440', 'ignorecommand', '']
['set', 'haproxy-badbots-1440', 'addaction', 'iptables-ipset-proto6-allports']
['multi-set', 'haproxy-badbots-1440', 'action', 'iptables-ipset-proto6-allports', [['actionstart', 'ipset create <ipmset> hash:ip timeout 600 <familyopt>\n<iptables> -I ban -m set --match-set <ipmset> src -j <blocktype>'], ['actionstart_on_demand', True], ['actionstop', '<iptables> -D ban -m set --match-set <ipmset> src -j <blocktype>\nipset flush <ipmset>\nipset destroy <ipmset>'], ['actionflush', 'ipset flush <ipmset>'], ['actionban', 'ipset add <ipmset> <ip> timeout <timeout> -exist'], ['actionunban', 'ipset del <ipmset> <ip> -exist'], ['name', 'haproxy-badbots-1440'], ['protocol', 'all'], ['bantime', '24h'], ['chain', 'ban'], ['actname', 'iptables-ipset-proto6-allports'], ['port', 'ssh'], ['blocktype', 'REJECT --reject-with icmp-port-unreachable'], ['returntype', 'RETURN'], ['lockingopt', '-w'], ['iptables', 'iptables <lockingopt>'], ['blocktype?family=inet6', 'REJECT --reject-with icmp6-port-unreachable'], ['iptables?family=inet6', 'ip6tables <lockingopt>'], ['default-timeout', '600'], ['timeout', '0'], ['timeout-bantime', '$([ "<bantime>" -le 2147483 ] && echo "<bantime>" || echo 0)'], ['ipmset', 'f2b-<name>'], ['familyopt', ''], ['ipmset?family=inet6', 'f2b-<name>-v6'], ['familyopt?family=inet6', 'family inet6']]]
['start', 'haproxy-badbots-1440']
```

#### Any customizations done to /etc/fail2ban/ configuration

Filter `haproxy-request.local`:

```
[Definition]

badbotscustom = Seekport Crawler|EmailCollector|WebEMailExtrac|TrackBack/1\.02|sogou music spider|SEMrushBot|BLEXBot
badbots = Atomic_Email_Hunter/4\.0|atSpider/1\.0|autoemailspider|bwh3_user_agent|China Local Browse 2\.6|ContactBot/0\.2|ContentSmartz|DataCha0s/2\.0|DBrowse 1\.4b|DBrowse 1\.4d|Demo Bot DOT 16b|Demo Bot Z 16b|DSurf15a 01|DSurf15a 71|DSurf15a 81|DSurf15a VA|EBrowse 1\.4b|Educate Search VxB|EmailSiphon|EmailSpider|EmailWolf 1\.00|ESurf15a 15|ExtractorPro|Franklin Locator 1\.8|FSurf15a 01|Full Web Bot 0416B|Full Web Bot 0516B|Full Web Bot 2816B|Guestbook Auto Submitter|Industry Program 1\.0\.x|ISC Systems iRc Search 2\.1|IUPUI Research Bot v 1\.9a|LARBIN-EXPERIMENTAL \(efp@gmx\.net\)|LetsCrawl\.com/1\.0 +http\://letscrawl\.com/|Lincoln State Web Browser|LMQueueBot/0\.2|LWP\:\:Simple/5\.803|Mac Finder 1\.0\.xx|MFC Foundation Class Library 4\.0|Microsoft URL Control - 6\.00\.8xxx|Missauga Locate 1\.0\.0|Missigua Locator 1\.9|Missouri College Browse|Mizzu Labs 2\.2|Mo College 1\.9|MVAClient|Mozilla/2\.0 \(compatible; NEWT ActiveX; Win32\)|Mozilla/3\.0 \(compatible; Indy Library\)|Mozilla/3\.0 \(compatible; scan4mail \(advanced version\) http\://www\.peterspages\.net/?scan4mail\)|Mozilla/4\.0 \(compatible; Advanced Email Extractor v2\.xx\)|Mozilla/4\.0 \(compatible; Iplexx Spider/1\.0 http\://www\.iplexx\.at\)|Mozilla/4\.0 \(compatible; MSIE 5\.0; Windows NT; DigExt; DTS Agent|Mozilla/4\.0 efp@gmx\.net|Mozilla/5\.0 \(Version\: xxxx Type\:xx\)|NameOfAgent \(CMS Spider\)|NASA Search 1\.0|Nsauditor/1\.x|PBrowse 1\.4b|PEval 1\.4b|Poirot|Port Huron Labs|Production Bot 0116B|Production Bot 2016B|Production Bot DOT 3016B|Program Shareware 1\.0\.2|PSurf15a 11|PSurf15a 51|PSurf15a VA|psycheclone|RSurf15a 41|RSurf15a 51|RSurf15a 81|searchbot admin@google\.com|ShablastBot 1\.0|snap\.com beta crawler v0|Snapbot/1\.0|Snapbot/1\.0 \(Snap Shots&#44; +http\://www\.snap\.com\)|sogou develop spider|Sogou Orion spider/3\.0\(+http\://www\.sogou\.com/docs/help/webmasters\.htm#07\)|sogou spider|Sogou web spider/3\.0\(+http\://www\.sogou\.com/docs/help/webmasters\.htm#07\)|sohu agent|SSurf15a 11 |TSurf15a 11|Under the Rainbow 2\.2|User-Agent\: Mozilla/4\.0 \(compatible; MSIE 6\.0; Windows NT 5\.1\)|VadixBot|WebVulnCrawl\.unknown/1\.0 libwww-perl/5\.803|Wells Search II|WEP Search 00

failregex = ^\s*\w+\s+haproxy\[\d+\]: <ADDR> .*(?:%(badbots)s|%(badbotscustom)s).*$

ignoreregex =

journalmatch = _SYSTEMD_UNIT=haproxy.service + _COMM=haproxy
```

`jail.local`:

```
# ...

# Default banning action
banaction = iptables-ipset-proto6-allports

# Specify chain where jumps would need to be added in iptables-* actions
chain = ban

#
# Action shortcuts. To be used to define action parameter
protocol_ipset6_ = all
action_ipset6_ = %(banaction)s[name=%(__name__)s, protocol="%(protocol_ipset6_)s", bantime="%(bantime)s", chain="%(chain)s"]
action = %(action_ipset6_)s

# ...

[haproxy-badbots-1440]

enabled  = true
filter   = haproxy-badbots
backend = %(haproxy_backend)s
maxretry = 6
bantime = 24h
findtime = 7h
```

#### Relevant parts of /var/log/fail2ban.log file:

```
2020-06-03 00:22:13,335 fail2ban.actions        [4264]: ERROR   Failed to execute ban jail 'haproxy-badbots-1440' action 'iptables-ipset-proto6-allports' info 'ActionInfo({'ip': '2a01:4f8:10b:dd5::2', 'family': 'inet6', 'ip-rev': '2.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.5.d.d.0.b.0.1.0.8.f.4.0.1.0.a.2.', 'ip-host': None, 'fid': '2a01:4f8:10b:dd5::2', 'failures': 6, 'time': 1591161732.783727, 'matches': '2020-06-03T00:21:18.480435sk0 haproxy[4164]: 2a01:4f8:10b:dd5::2 200 [03/Jun/2020:00:21:18.162] https_frontend~ www_backend/sk1 1/0/0/316/317 200 13117 - - --VN 1598/1598/6/6/0 0/0 Mozilla/5.0 (compatible; BLEXBot/1.0; +http://webmeup-crawler.co {www.REDACTED.org|Mozilla/5.0 (compatible; BLEXBot/1.0; +http://webmeup-crawler.co} "GET /membership/member-news/bringing-together-community-communitea/may-8-2020 HTTP/1.1" Mozilla/5.0 (compatible; BLEXBot/1.0; +http://webmeup-crawler.co\n2020-06-03T00:21:30.928395sk0 haproxy[4164]: 2a01:4f8:10b:dd5::2 200 [03/Jun/2020:00:21:30.556] https_frontend~ www_backend/sk1 1/0/1/369/371 200 13662 - - --VN 1581/1581/1/1/0 0/0 Mozilla/5.0 (compatible; BLEXBot/1.0; +http://webmeup-crawler.co {www.REDACTED.org|Mozilla/5.0 (compatible; BLEXBot/1.0; +http://webmeup-crawler.co} "GET /membership/member-news/marc-ensures-health-and-safety-staff-and-patients/may-8-2020 HTTP/1.1" Mozilla/5.0 (compatible; BLEXBot/1.0; +http://webmeup-crawler.co\n2020-06-03T00:21:43.289848sk0 haproxy[4164]: 2a01:4f8:10b:dd5::2 200 [03/Jun/2020:00:21:42.959] https_frontend~ www_backend/sk1 1/0/0/330/331 200 13101 - - --VN 1574/1574/3/3/0 0/0 Mozilla/5.0 (compatible; BLEXBot/1.0; +http://webmeup-crawler.co {www.REDACTED.org|Mozilla/5.0 (compatible; BLEXBot/1.0; +http://webmeup-crawler.co} "GET /membership/member-news/REDACTED-interviews-dr-tom-britton-ceo-president-gateway-foundation/may-8-2020 HTTP/1.1" Mozilla/5.0 (compatible; BLEXBot/1.0; +http://webmeup-crawler.co\n2020-06-03T00:21:58.103597sk0 haproxy[4164]: 2a01:4f8:10b:dd5::2 301 [03/Jun/2020:00:21:58.102] https_frontend~ www_backend/sk1 0/0/1/0/1 301 532 - - --NI 1575/1575/2/2/0 0/0 Mozilla/5.0 (compatible; BLEXBot/1.0; +http://webmeup-crawler.co {REDACTED.org|Mozilla/5.0 (compatible; BLEXBot/1.0; +http://webmeup-crawler.co} "GET / HTTP/1.1" Mozilla/5.0 (compatible; BLEXBot/1.0; +http://webmeup-crawler.co\n2020-06-03T00:21:58.620521sk0 haproxy[4164]: 2a01:4f8:10b:dd5::2 200 [03/Jun/2020:00:21:58.605] https_frontend~ www_backend/sk1 0/0/1/14/15 200 12049 - - --NI 1574/1574/1/1/0 0/0 Mozilla/5.0 (compatible; BLEXBot/1.0; +http://webmeup-crawler.co {www.REDACTED.org|Mozilla/5.0 (compatible; BLEXBot/1.0; +http://webmeup-crawler.co} "GET / HTTP/1.1" Mozilla/5.0 (compatible; BLEXBot/1.0; +http://webmeup-crawler.co\n2020-06-03T00:22:12.783727sk0 haproxy[4164]: 2a01:4f8:10b:dd5::2 301 [03/Jun/2020:00:22:12.736] https_frontend~ www_backend/sk1 0/0/1/46/47 301 555 - - --VN 1562/1562/2/2/0 0/0 Mozilla/5.0 (compatible; BLEXBot/1.0; +http://webmeup-crawler.co {www.REDACTED.org|Mozilla/5.0 (compatible; BLEXBot/1.0; +http://webmeup-crawler.co} "GET /about-REDACTED/ethics-policy/ HTTP/1.1" Mozilla/5.0 (compatible; BLEXBot/1.0; +http://webmeup-crawler.co', 'restored': 0, 'F-*': {'matches': [('', '2020-06-03T00:21:18.480435', 'sk0 haproxy[4164]: 2a01:4f8:10b:dd5::2 200 [03/Jun/2020:00:21:18.162] https_frontend~ www_backend/sk1 1/0/0/316/317 200 13117 - - --VN 1598/1598/6/6/0 0/0 Mozilla/5.0 (compatible; BLEXBot/1.0; +http://webmeup-crawler.co {www.REDACTED.org|Mozilla/5.0 (compatible; BLEXBot/1.0; +http://webmeup-crawler.co} "GET /membership/member-news/bringing-together-community-communitea/may-8-2020 HTTP/1.1" Mozilla/5.0 (compatible; BLEXBot/1.0; +http://webmeup-crawler.co'), '2020-06-03T00:21:30.928395sk0 haproxy[4164]: 2a01:4f8:10b:dd5::2 200 [03/Jun/2020:00:21:30.556] https_frontend~ www_backend/sk1 1/0/1/369/371 200 13662 - - --VN 1581/1581/1/1/0 0/0 Mozilla/5.0 (compatible; BLEXBot/1.0; +http://webmeup-crawler.co {www.REDACTED.org|Mozilla/5.0 (compatible; BLEXBot/1.0; +http://webmeup-crawler.co} "GET /membership/member-news/marc-ensures-health-and-safety-staff-and-patients/may-8-2020 HTTP/1.1" Mozilla/5.0 (compatible; BLEXBot/1.0; +http://webmeup-crawler.co', '2020-06-03T00:21:43.289848sk0 haproxy[4164]: 2a01:4f8:10b:dd5::2 200 [03/Jun/2020:00:21:42.959] https_frontend~ www_backend/sk1 1/0/0/330/331 200 13101 - - --VN 1574/1574/3/3/0 0/0 Mozilla/5.0 (compatible; BLEXBot/1.0; +http://webmeup-crawler.co {www.REDACTED.org|Mozilla/5.0 (compatible; BLEXBot/1.0; +http://webmeup-crawler.co} "GET /membership/member-news/REDACTED-interviews-dr-tom-britton-ceo-president-gateway-foundation/may-8-2020 HTTP/1.1" Mozilla/5.0 (compatible; BLEXBot/1.0; +http://webmeup-crawler.co', '2020-06-03T00:21:58.103597sk0 haproxy[4164]: 2a01:4f8:10b:dd5::2 301 [03/Jun/2020:00:21:58.102] https_frontend~ www_backend/sk1 0/0/1/0/1 301 532 - - --NI 1575/1575/2/2/0 0/0 Mozilla/5.0 (compatible; BLEXBot/1.0; +http://webmeup-crawler.co {REDACTED.org|Mozilla/5.0 (compatible; BLEXBot/1.0; +http://webmeup-crawler.co} "GET / HTTP/1.1" Mozilla/5.0 (compatible; BLEXBot/1.0; +http://webmeup-crawler.co', '2020-06-03T00:21:58.620521sk0 haproxy[4164]: 2a01:4f8:10b:dd5::2 200 [03/Jun/2020:00:21:58.605] https_frontend~ www_backend/sk1 0/0/1/14/15 200 12049 - - --NI 1574/1574/1/1/0 0/0 Mozilla/5.0 (compatible; BLEXBot/1.0; +http://webmeup-crawler.co {www.REDACTED.org|Mozilla/5.0 (compatible; BLEXBot/1.0; +http://webmeup-crawler.co} "GET / HTTP/1.1" Mozilla/5.0 (compatible; BLEXBot/1.0; +http://webmeup-crawler.co', '2020-06-03T00:22:12.783727sk0 haproxy[4164]: 2a01:4f8:10b:dd5::2 301 [03/Jun/2020:00:22:12.736] https_frontend~ www_backend/sk1 0/0/1/46/47 301 555 - - --VN 1562/1562/2/2/0 0/0 Mozilla/5.0 (compatible; BLEXBot/1.0; +http://webmeup-crawler.co {www.REDACTED.org|Mozilla/5.0 (compatible; BLEXBot/1.0; +http://webmeup-crawler.co} "GET /about-REDACTED/ethics-policy/ HTTP/1.1" Mozilla/5.0 (compatible; BLEXBot/1.0; +http://webmeup-crawler.co'], 'failures': 6, 'ip6': '2a01:4f8:10b:dd5::2'}, 'ipmatches': '2020-06-03T00:21:18.480435sk0 haproxy[4164]: 2a01:4f8:10b:dd5::2 200 [03/Jun/2020:00:21:18.162] https_frontend~ www_backend/sk1 1/0/0/316/317 200 13117 - - --VN 1598/1598/6/6/0 0/0 Mozilla/5.0 (compatible; BLEXBot/1.0; +http://webmeup-crawler.co {www.REDACTED.org|Mozilla/5.0 (compatible; BLEXBot/1.0; +http://webmeup-crawler.co} "GET /membership/member-news/bringing-together-community-communitea/may-8-2020 HTTP/1.1" Mozilla/5.0 (compatible; BLEXBot/1.0; +http://webmeup-crawler.co\n2020-06-03T00:21:30.928395sk0 haproxy[4164]: 2a01:4f8:10b:dd5::2 200 [03/Jun/2020:00:21:30.556] https_frontend~ www_backend/sk1 1/0/1/369/371 200 13662 - - --VN 1581/1581/1/1/0 0/0 Mozilla/5.0 (compatible; BLEXBot/1.0; +http://webmeup-crawler.co {www.REDACTED.org|Mozilla/5.0 (compatible; BLEXBot/1.0; +http://webmeup-crawler.co} "GET /membership/member-news/marc-ensures-health-and-safety-staff-and-patients/may-8-2020 HTTP/1.1" Mozilla/5.0 (compatible; BLEXBot/1.0; +http://webmeup-crawler.co\n2020-06-03T00:21:43.289848sk0 haproxy[4164]: 2a01:4f8:10b:dd5::2 200 [03/Jun/2020:00:21:42.959] https_frontend~ www_backend/sk1 1/0/0/330/331 200 13101 - - --VN 1574/1574/3/3/0 0/0 Mozilla/5.0 (compatible; BLEXBot/1.0; +http://webmeup-crawler.co {www.REDACTED.org|Mozilla/5.0 (compatible; BLEXBot/1.0; +http://webmeup-crawler.co} "GET /membership/member-news/REDACTED-interviews-dr-tom-britton-ceo-president-gateway-foundation/may-8-2020 HTTP/1.1" Mozilla/5.0 (compatible; BLEXBot/1.0; +http://webmeup-crawler.co\n2020-06-03T00:21:58.103597sk0 haproxy[4164]: 2a01:4f8:10b:dd5::2 301 [03/Jun/2020:00:21:58.102] https_frontend~ www_backend/sk1 0/0/1/0/1 301 532 - - --NI 1575/1575/2/2/0 0/0 Mozilla/5.0 (compatible; BLEXBot/1.0; +http://webmeup-crawler.co {REDACTED.org|Mozilla/5.0 (compatible; BLEXBot/1.0; +http://webmeup-crawler.co} "GET / HTTP/1.1" Mozilla/5.0 (compatible; BLEXBot/1.0; +http://webmeup-crawler.co\n2020-06-03T00:21:58.620521sk0 haproxy[4164]: 2a01:4f8:10b:dd5::2 200 [03/Jun/2020:00:21:58.605] https_frontend~ www_backend/sk1 0/0/1/14/15 200 12049 - - --NI 1574/1574/1/1/0 0/0 Mozilla/5.0 (compatible; BLEXBot/1.0; +http://webmeup-crawler.co {www.REDACTED.org|Mozilla/5.0 (compatible; BLEXBot/1.0; +http://webmeup-crawler.co} "GET / HTTP/1.1" Mozilla/5.0 (compatible; BLEXBot/1.0; +http://webmeup-crawler.co\n2020-06-03T00:22:12.783727sk0 haproxy[4164]: 2a01:4f8:10b:dd5::2 301 [03/Jun/2020:00:22:12.736] https_frontend~ www_backend/sk1 0/0/1/46/47 301 555 - - --VN 1562/1562/2/2/0 0/0 Mozilla/5.0 (compatible; BLEXBot/1.0; +http://webmeup-crawler.co {www.REDACTED.org|Mozilla/5.0 (compatible; BLEXBot/1.0; +http://webmeup-crawler.co} "GET /about-REDACTED/ethics-policy/ HTTP/1.1" Mozilla/5.0 (compatible; BLEXBot/1.0; +http://webmeup-crawler.co','ipjailmatches': '2020-06-03T00:21:18.480435sk0 haproxy[4164]: 2a01:4f8:10b:dd5::2 200 [03/Jun/2020:00:21:18.162] https_frontend~ www_backend/sk1 1/0/0/316/317 200 13117 - - --VN 1598/1598/6/6/0 0/0 Mozilla/5.0 (compatible; BLEXBot/1.0; +http://webmeup-crawler.co {www.REDACTED.org|Mozilla/5.0 (compatible; BLEXBot/1.0; +http://webmeup-crawler.co} "GET /membership/member-news/bringing-together-community-communitea/may-8-2020 HTTP/1.1" Mozilla/5.0 (compatible; BLEXBot/1.0; +http://webmeup-crawler.co\n2020-06-03T00:21:30.928395sk0 haproxy[4164]: 2a01:4f8:10b:dd5::2 200 [03/Jun/2020:00:21:30.556] https_frontend~ www_backend/sk1 1/0/1/369/371 200 13662 - - --VN 1581/1581/1/1/0 0/0 Mozilla/5.0 (compatible; BLEXBot/1.0; +http://webmeup-crawler.co {www.REDACTED.org|Mozilla/5.0 (compatible; BLEXBot/1.0; +http://webmeup-crawler.co} "GET /membership/member-news/marc-ensures-health-and-safety-staff-and-patients/may-8-2020 HTTP/1.1" Mozilla/5.0 (compatible; BLEXBot/1.0; +http://webmeup-crawler.co\n2020-06-03T00:21:43.289848sk0 haproxy[4164]: 2a01:4f8:10b:dd5::2 200 [03/Jun/2020:00:21:42.959] https_frontend~ www_backend/sk1 1/0/0/330/331 200 13101 - - --VN 1574/1574/3/3/0 0/0 Mozilla/5.0 (compatible; BLEXBot/1.0; +http://webmeup-crawler.co {www.REDACTED.org|Mozilla/5.0 (compatible; BLEXBot/1.0; +http://webmeup-crawler.co} "GET /membership/member-news/REDACTED-interviews-dr-tom-britton-ceo-president-gateway-foundation/may-8-2020 HTTP/1.1" Mozilla/5.0 (compatible; BLEXBot/1.0; +http://webmeup-crawler.co\n2020-06-03T00:21:58.103597sk0 haproxy[4164]: 2a01:4f8:10b:dd5::2 301 [03/Jun/2020:00:21:58.102] https_frontend~ www_backend/sk1 0/0/1/0/1 301 532 - - --NI 1575/1575/2/2/0 0/0 Mozilla/5.0 (compatible; BLEXBot/1.0; +http://webmeup-crawler.co {REDACTED.org|Mozilla/5.0 (compatible; BLEXBot/1.0; +http://webmeup-crawler.co} "GET / HTTP/1.1" Mozilla/5.0 (compatible; BLEXBot/1.0; +http://webmeup-crawler.co\n2020-06-03T00:21:58.620521sk0 haproxy[4164]: 2a01:4f8:10b:dd5::2 200 [03/Jun/2020:00:21:58.605] https_frontend~ www_backend/sk1 0/0/1/14/15 200 12049 - - --NI 1574/1574/1/1/0 0/0 Mozilla/5.0 (compatible; BLEXBot/1.0; +http://webmeup-crawler.co {www.REDACTED.org|Mozilla/5.0 (compatible; BLEXBot/1.0; +http://webmeup-crawler.co} "GET / HTTP/1.1" Mozilla/5.0 (compatible; BLEXBot/1.0; +http://webmeup-crawler.co\n2020-06-03T00:22:12.783727sk0 haproxy[4164]: 2a01:4f8:10b:dd5::2 301 [03/Jun/2020:00:22:12.736] https_frontend~ www_backend/sk1 0/0/1/46/47 301 555 - - --VN 1562/1562/2/2/0 0/0 Mozilla/5.0 (compatible; BLEXBot/1.0; +http://webmeup-crawler.co {www.REDACTED.org|Mozilla/5.0 (compatible; BLEXBot/1.0; +http://webmeup-crawler.co} "GET /about-REDACTED/ethics-policy/ HTTP/1.1" Mozilla/5.0 (compatible; BLEXBot/1.0; +http://webmeup-crawler.co', 'ipfailures': 6, 'ipjailfailures': 6})': unsupported operand type(s) for +: 'float' and 'str'
```

**Comment by sebres** [comment]
> Regular expression not terminating on '$'

I quite not understand what you mean with "terminating" here. `$` is an end-anchor and mean the RE **must** parse to end of input string.

By the way part of RE `.*$` (anchor after catch-all) does not make sense at all. You can remove this and it would change nothing:
```
$ python -c 'import re; print(bool(re.search("\d+.*$", "test 123 string")))'
True
$ python -c 'import re; print(bool(re.search("\d+", "test 123 string")))'
True

$ python -c 'import re; print(bool(re.search("\d+$", "test 123 string")))'
False
$ python -c 'import re; print(bool(re.search("\d+ \w+$", "test 123 string")))'
True
```
Anchor makes sense only if you would specify the RE before anchor precise as possible (and catch-all neglecting that).

But that is not your issue basically...

> ` unsupported operand type(s) for +: 'float' and 'str'`

This is probably the error either.
And it is already fixed, see https://github.com/fail2ban/fail2ban/issues/2241#issuecomment-425048992 (released as 0.10.4).

**Comment by johnramsden** [comment]
Thank you.
